tachyon/math/polynomials/univariate/univariate_polynomial_ops.h

```diff
@@ -383,31 +383,40 @@ class UnivariatePolynomialOp<UnivariateDenseCoefficients<F, MaxDegree>> {
   }
 
   template <typename DOrS>
-  static UnivariatePolynomial<D> Mod(const UnivariatePolynomial<D>& self,
-                                     const UnivariatePolynomial<DOrS>& other) {
+  constexpr static std::optional<UnivariatePolynomial<D>> Mod(
+      const UnivariatePolynomial<D>& self,
+      const UnivariatePolynomial<DOrS>& other) {
     DivResult<UnivariatePolynomial<D>> result;
-    CHECK(Divide(self, other, result));
-    result.remainder.coefficients_.RemoveHighDegreeZeros();
-    return result.remainder;
+    if (LIKELY(Divide(self, other, result))) {
+      result.remainder.coefficients_.RemoveHighDegreeZeros();
+      return result.remainder;
+    }
+    LOG_IF_NOT_GPU(ERROR) << "Division by zero attempted";
```

Comment:
To provide better error logging in these cases, what about changing the error log to "Division by zero attempted by mod"? (+ for all other cases in this commit if you agree)